### PR TITLE
fix not completing the writing of the JsonRPC response

### PR DESCRIPTION
**Pull request by smartprogrammer93**
## Changes

- fix not completing the writing

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

### CODE REVIEW

File: src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs [hunk 1]
```diff
@@ -42,6 +43,12 @@ public void Advance(int count)
     public Memory<byte> GetMemory(int sizeHint = 0) => _writer.GetMemory(sizeHint);
 
     public Span<byte> GetSpan(int sizeHint = 0) => _writer.GetSpan(sizeHint);
+
+    public ValueTask CompleteAsync(Exception? exception = null)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:47]: what is the purpose of Exception here?

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:47]: it is an overrided method

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs (hunk 1 above)
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:47]: image

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:47]: I do not see any overrides here, you are implementing `ICountingBufferWriter`

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:47]: two classes are implementing `ICountingBufferWriter` and one of them has this method as an override to the one in `PipeWriter`

**Pull request merged**